### PR TITLE
update chap on csi-driver upgrade

### PR DESCRIPTION
**Pull request by raunakkumar**
Problem: we don't manage CRDs between releases with helm.
This leaves the hpenodeinfo crd as is between releases for chap updates
Implementation: update chap just like other hpenodeinfo params

Signed-off-by: Raunak <raunak.kumar@hpe.com>

### CODE REVIEW

File: pkg/flavor/kubernetes/flavor.go [hunk 1]
```diff
@@ -281,6 +293,14 @@ func getWwpnsFromNode(node *model.Node) []string {
 	return wwpns
 }
 
+func getChapUserFromEnvironment() string {
+	return os.Getenv("CHAP_USER")
```
Comment (sneharai4) [inline pkg/flavor/kubernetes/flavor.go:297]: Can we use here, chapUserEnvKey const?

File: pkg/flavor/kubernetes/flavor.go [hunk 2]
```diff
@@ -281,6 +293,14 @@ func getWwpnsFromNode(node *model.Node) []string {
...
+}
+
+func getChapPasswordFromEnvironment() string {
+	return os.Getenv("CHAP_PASSWORD")
```
Comment (sneharai4) [inline pkg/flavor/kubernetes/flavor.go:301]: Can we use here, chapPasswordEnvKey const?

File: pkg/flavor/kubernetes/flavor.go (hunk 1 above)
Comment (raunakkumar) [reply pkg/flavor/kubernetes/flavor.go:297]: Importing constants creates a cyclic import dependency.

Comment (sneharai4) [reply pkg/flavor/kubernetes/flavor.go:297]: Can we update https://github.com/hpe-storage/csi-driver/blob/master/pkg/driver/node_server.go#L2059
https://github.com/hpe-storage/csi-driver/blob/master/pkg/driver/node_server.go#L2060 to call the new method?

**Review by sneharai4 — APPROVED** [review]

File: pkg/flavor/kubernetes/flavor.go [hunk 3]
```diff
@@ -224,12 +234,24 @@ func (flavor *Flavor) LoadNodeInfo(node *model.Node) (string, error) {
...
 
+		chapUser := flavor.GetChapUserFromEnvironment()
+		if !reflect.DeepEqual(nodeInfo.Spec.ChapUser, chapUser) {
+			nodeInfo.Spec.ChapUser = chapUser
+			updateNodeRequired = true
+		}
+
+		chapPassword := flavor.GetChapPasswordFromEnvironment()
+		if !reflect.DeepEqual(nodeInfo.Spec.ChapPassword, chapPassword) {
+			nodeInfo.Spec.ChapPassword = chapPassword
```
Comment (sneharai4) [inline pkg/flavor/kubernetes/flavor.go:245]: Do we need to update chap with encoded value like we do here https://github.com/hpe-storage/csi-driver/blob/master/pkg/flavor/kubernetes/flavor.go#L250?

Comment (raunakkumar) [reply pkg/flavor/kubernetes/flavor.go]: Yes, thanks for catching this.

**Review by suneeth51 — APPROVED** [review]

**Pull request merged**